app/controllers/auth_controller.rb

```diff
@@ -17,20 +20,25 @@ def sign_in_action
       sign_in :user, user
       render json: true
     else
-      error! "Username or password incorrect", 401
+      error! 'Username or password incorrect', 401
     end
   end
 
   def sign_up_action
-    return error! "Missing parameters", 400 unless [:username, :email, :password].none? { |x| params[x].blank? }
-    return error! "User with that email already exists", 409 if User.exists?(['lower(email) = ?', params[:email].downcase])
-    return error! "User with that name already exists", 409 if User.exists?(['lower(name) = ?', params[:username].downcase])
+    return error! 'Missing parameters', 400 unless %i(username email password)
+                                                   .none? do |x|
+                                                     params[x].blank?
+                                                   end
+    return error! 'User with that email already exists', 409 if User.exists?(
```

Comment:
These should probably get offloaded to model methods, but if you don't wanna bother right now this works well enough